src/model.py

```diff
@@ -155,9 +157,28 @@ def init_engine(self):
         self.setup_lora()
 
         # Create an AsyncLLMEngine from the config from JSON
-        self.llm_engine = AsyncLLMEngine.from_engine_args(
-            AsyncEngineArgs(**self.vllm_engine_config)
-        )
+        aync_engine_args = AsyncEngineArgs(**self.vllm_engine_config)
+        self.llm_engine = AsyncLLMEngine.from_engine_args(aync_engine_args)
+
+        # Create vLLM custom metrics
+        if (
+            "REPORT_CUSTOM_METRICS" in self.model_config["parameters"]
+            and self.model_config["parameters"]["REPORT_CUSTOM_METRICS"]["string_value"]
+            == "yes"
+        ):
```

Comment:
potentially we can also check if `disable_log_stats` is `true`

Reply:
Nice catch. If `FORCE_CPU_ONLY_INPUT_TENSORS = true` but `disable_log_stats=true`, `add_logger()` throws exception. Test added.

Reply:
you mean `REPORT_CUSTOM_METRICS` not `FORCE_CPU_ONLY_INPUT_TENSORS` ?

Reply:
Sorry. Yes I meant `REPORT_CUSTOM_METRICS`.